### PR TITLE
Fix zoom in drilldown for FKs

### LINKED ISSUE #23293: Drill-through to questions with add columns (implicit join) causes no filtering to be defined
**Describe the bug**
Drill-through to questions with add columns (implicitly joined) causes no filtering to be defined.

Workaround: Use explicit join in the first question (join Products table instead of step 2), or use [remapping](https://www.metabase.com/docs/latest/administration-guide/03-metadata-editing.html#remapping-column-values).

**To Reproduce**
1. Question > Sample > Orders
2. Visualization Settings > Visible columns and More columns > Remove Orders.ProductID, and add Products.Category
image
3. Save question as "Q1"
4. Question > Saved Question > "Q1" - summarize Count by Category
image
5. Save question as "Q2" (try drill-through on the question, works as expected)
6. Add "Q2" to a dashboard, click the bars to do a "View these Q1"
image
7. Shows all results instead of filtered accordingly to the drill-through
image

**Information about your Metabase Installation:**
Tested 0.43.2

**Additional context**
Likely done this way because of https://github.com/metabase/metabase/issues/15163#issuecomment-990132061.

### LINKED ISSUE #27380: Doing "Zoom in" from dashboard drops fields from joined tables
**Describe the bug**
Doing "Zoom in" drill from dashboard drops fields from joined tables used in the group-by. This does not happen when using drag-to-zoom.

Possible workaround is to click the question title on the dashboard, or use drag-to-zoom on the dashboard.

**To Reproduce**
1. Question > Sample > Orders
2. Summarize Count grouped by **Products**.CreatedAt:Month (doesn't matter if Products is implicit or explicit joined).
3. Save question and add to dashboard
4. Click one of the line points and "Zoom in" - shows invalid chart (not a very helpful error) and looking in Notebook will show that it has dropped the group-by field.

https://user-images.githubusercontent.com/1447303/209137652-a4aaa6de-a293-49d7-acf4-675995ba35d3.mp4

**Information about your Metabase Installation:**
Tested 0.45.1

**Additional context**
Feels related to #23293

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/27380
Fixes #23293

How to test:
- Follow the steps from the issue
- It can be a bit tricky, so I would recommend relying on the unskipped CI to pass

Screenshot 2023-02-02 at 15 53 19

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28022)

### CODE REVIEW

File: frontend/src/metabase-lib/queries/utils/drilldown.js [hunk 1]
```diff
@@ -449,7 +449,7 @@ function columnToFieldDimension(column, metadata) {
     return;
   }
 
-  const dimension = new FieldDimension(column.id, null, metadata);
+  const dimension = FieldDimension.parseMBQL(column.field_ref, metadata);
```
Comment (ranquild) [inline frontend/src/metabase-lib/queries/utils/drilldown.js:452]: The issue was that `source-field` option from `column` was previously dropped

File: frontend/test/metabase/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js [hunk 2]
```diff
@@ -40,10 +40,10 @@ describe.skip("issue 27380", () => {
 
     // Graph should still exist
     // Let's check only the y-axis label
-    cy.get("y-axis-label").invoke("text").should("eq", "Count");
+    cy.get(".y-axis-label").invoke("text").should("eq", "Count");
 
     cy.icon("notebook").click();
     cy.findByText("Pick a column to group by").should("not.exist");
-    cy.findByText(/Products? → Created At: Month/);
+    cy.findByText("Count by Product → Created At: Week");
```
Comment (ranquild) [inline frontend/test/metabase/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js:47]: @nemanjaglumac not sure if another title was intended. Could you double check?

Comment (ranquild) [reply frontend/test/metabase/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js:47]: At least currently we override `temporal-unit` from the original breakout to match the default one for the column https://github.com/metabase/metabase/blob/5e839b3556de06070e0eaf0bcfce915c37aa5f03/frontend/src/metabase-lib/queries/utils/drilldown.js#L454. I don't think we should change it here. I'd like this PR to address only foreign keys not working with `zoom in` drilldown. WDYT?

**Review by kulyk — APPROVED** [review]

File: frontend/test/metabase/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js:47]: Tbh, I don't remember.
I was following the repro steps and in the last step it says: "The group by column gets dropped". The group by column was "Product -> Created At: Month", so I guess that was my reasoning.

**Review by nemanjaglumac — APPROVED** [review]
Thank you for fixing/improving the E2E repro.
This now works as expected.

**Comment by nemanjaglumac** [comment]
Bonus points - this seems to fix #23293 as well.
Just tested locally.

**Comment by nemanjaglumac** [comment]
@ranquild backport?

**Pull request merged**